Updates for webpacker_lite 2.0

justin808 opened the pull request:
See https://github.com/shakacode/webpacker_lite/pull/9

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/shakacode/react_on_rails/844)
<!-- Reviewable:end -->

ledestin commented on webpackConfigLoader.js:
```diff
@@ -1,27 +1,32 @@
 
 const configLoader = (configPath) => {
-  const paths = safeLoad(readFileSync(join(configPath, 'paths.yml'), 'utf8'))[env.NODE_ENV];
+  const configuration = safeLoad(readFileSync(join(configPath, 'webpacker_lite.yml'), 'utf8'))[env.NODE_ENV];
```
At least I'd consider extracting the absolute file name as a constant.

ledestin commented on webpackConfigLoader.js:
```diff
@@ -1,27 +1,32 @@
-    devServer.enabled = true;
+  const manifest = webpackOutputPath.manifest;
+  let hotReloadingServerEnabled = false;
+  if (env.HOT_RELOADING === 'TRUE' || env.HOT_RELOADING === 'YES' ||
```
Consider extracting to `env.hotReloadingEnabled` to hide `TRUE` and `YES` details.

ledestin commented on webpackConfigLoader.js:
```diff
@@ -1,27 +1,32 @@
 const { safeLoad } = require('js-yaml');
 const { readFileSync } = require('fs');
 
 const configLoader = (configPath) => {
```
`configLoader` isn't used anywhere, it seems?

justin808 commented:
Review status: 0 of 19 files reviewed at latest revision, 4 unresolved discussions, some commit checks broke.

---

*[spec/dummy/Procfile.hot, line 6 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/844#-KkhKRBQQaRdH68enl-D:-KkhKRBRi9eA72P1NxmC:boujjb) ([raw file](https://github.com/shakacode/react_on_rails/blob/8a3279b7c32eea58cf40f9fc37406065e4b19c63/spec/dummy/Procfile.hot#L6)):*
> ```
> 
> # Run the hot reload server for client development
> hot-assets: HOT_RAILS_PORT=3500 REACT_ON_RAILS_ENV=HOT yarn run hot-assets
> ```

@conturbo be sure to put this in the migration doc

1. no use of HOT_RAILS_PORT env
2. change the to HOT_RELOADING=TRUE

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/844)*
<!-- Sent from Reviewable.io -->

justin808 commented:
Review status: 0 of 28 files reviewed at latest revision, 5 unresolved discussions, some commit checks broke.

---

*[webpackConfigLoader.js, line 6 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/844#-KkLBqh38t7wEFetl3zl-r1-6:-KkhihD_zz5FTK1XoeeU:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/c49621a73705ce3dc79dfd5523b60f086247b4e2/webpackConfigLoader.js#L6)):*
<details><summary><i>Previously, ledestin (Dmitry Maksyoma) wrote…</i></summary><blockquote>

`configLoader` isn't used anywhere, it seems?
</blockquote></details>

Done.

---

*[webpackConfigLoader.js, line 7 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/844#-KkLBqh38t7wEFetl3zl-r1-7:-KkhiiJAFeukUDHH-KE4:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/c49621a73705ce3dc79dfd5523b60f086247b4e2/webpackConfigLoader.js#L7)):*
<details><summary><i>Previously, ledestin (Dmitry Maksyoma) wrote…</i></summary><blockquote>

At least I'd consider extracting the absolute file name as a constant.
</blockquote></details>

Done.

---

*[spec/dummy/Gemfile, line 46 at r3](https://reviewable.io:443/reviews/shakacode/react_on_rails/844#-KkhqLj3usGMUCr-rljh:-KkhqLj3usGMUCr-rlji:b-9icfbf) ([raw file](https://github.com/shakacode/react_on_rails/blob/4c439fec4b832b59d10710f172db873f180281f3/spec/dummy/Gemfile#L46)):*
> ```Ruby
> gem "react_on_rails", path: "../.."
> 
> gem "webpacker_lite", path: "../../../webpacker_lite"
> ```

remove later

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/844)*
<!-- Sent from Reviewable.io -->

Conturbo commented on lib/generators/react_on_rails/templates/base/base/client/webpack.config.js:
```diff
@@ -26,7 +26,7 @@ const config = {
     // If you use named entries (Object rather than Array), then remove the "bundle" from this
     // "filename" and add "bundle" to your named entries.
-    filename: 'webpack-bundle-[hash].js',
+    filename: '/webpack-bundle-[hash].js',
```
This may be a mistake, i'll check it again.

Conturbo commented on spec/dummy/client/webpack.client.rails.build.config.js:
```diff
@@ -25,7 +25,7 @@ module.exports = merge(config, {
 
     // Leading slash is necessary
-    publicPath: '/' + webpackPublicOutputDir,
+    publicPath: '/' + webpackPublicOutputDir + '/',
```
This was a change @robwise and I made bc not all leading slashes were added in all the right places. (Only in images).  Thus, we removed the leading slashes before images in webpack.common.config.js

Conturbo commented on spec/dummy/client/webpack.common.config.js:
```diff
@@ -8,9 +8,7 @@ const assetLoaderRules = [
-
-        // Leading slash is needed
-        name: '/images/[name]-[hash].[ext]',
+        name: 'images/[name]-[hash].[ext]',
```
We removed the leading slash because we already appended it as a trailing slash in webpack.client.rails.build.config.js as explained.

Conturbo commented on spec/dummy/client/webpack.common.config.js:
```diff
@@ -19,8 +17,7 @@ const assetLoaderRules = [
       options: {
-        // Leading slash is 100% needed
-        name: '/images/[name]-[hash].[ext]',
+        name: 'images/[name]-[hash].[ext]',
```
We removed the leading slash because we already appended it as a trailing slash in webpack.client.rails.build.config.js as explained.

Conturbo commented on spec/dummy/client/webpack.server.rails.build.config.js:
```diff
@@ -22,7 +22,7 @@ module.exports = {
 
     // This needs a leading slash.
-    publicPath: '/' + webpackPublicOutputDir,
+    publicPath: '/' + webpackPublicOutputDir + '/',
```
This was a change @robwise and I made bc not all leading slashes were added in all the right places. (Only in images). Thus, we removed the leading slashes before images in webpack.common.config.js

Conturbo commented on spec/dummy/spec/features/integration_spec.rb:
```diff
@@ -158,6 +158,7 @@ def change_text_expect_dom_selector(dom_selector)
   background { visit "/deferred_render_with_server_rendering" }
   scenario "clicking on async route causes async component to be fetched" do
     header_text = page.find(:css, "h1").text
+
```
remove

justin808 commented:
```
cd /home/travis/build/shakacode/react_on_rails/gen-examples/examples/basic-server-rendering && COVERAGE=true TEST_ENV_COMMAND_NAME="basic-server-rendering" bundle exec rspec
Hello World
====> React On Rails: Checking public/webpack/test for outdated/missing bundles
Detected are the following stale generated files:
public/webpack/test/webpack-bundle.js
React on Rails will ensure your JavaScript generated files are up to date, using your
/client level package.json `yarn run build:test` command.
Building Webpack assets...
Completed building Webpack assets.
You're running an old version of PhantomJS, update to >= 2.1.1 for a better experience.
You specified server rendering JS file: public/webpack/test/webpack-bundle.js, but it cannot be read. You may set the server_bundle_js_file in your configuration to be "" to avoid this warning
  the hello world example works (FAILED - 1)
Failures:
  1) Hello World the hello world example works
     Failure/Error: <%= react_component("HelloWorld", props: @hello_world_props, prerender: true) %>
     
     ActionView::Template::Error:
       ERROR in SERVER PRERENDERING
       Encountered error: "ReferenceError: ReactOnRails is not defined"
       when prerendering HelloWorld with props: {"name":"Stranger"}
       js_code was:
       (function() {
         var railsContext = {"inMailer":false,"i18nLocale":"en","i18nDefaultLocale":"en","href":"http://127.0.0.1:40384/hello_world","location":"/hello_world","scheme":"http","host":"127.0.0.1","port":40384,"pathname":"/hello_world","search":null,"httpAcceptLanguage":"en-US,*","serverSide":true};     
```

robwise commented on spec/dummy/spec/features/rails_context_spec.rb:
```diff
@@ -30,6 +30,7 @@
         i18nLocale: "en",
         i18nDefaultLocale: "en",
         httpAcceptLanguage: http_accept_language,
+        # railsEnv: Rails.env,
```
commented-out code

justin808 commented:
@robwise, @samnang 
 
```ruby
    # TODO: conturbo Write Test for this, with BOTH webpacker_lite installed and not
    def self.server_bundle_js_file_path
      server_bundle_js_file = ReactOnRails.configuration.server_bundle_js_file

      if ActionController::Base.helpers.respond_to?(:asset_pack_path)
        # If using webpacker_lite gem
        public_subdir = ActionController::Base.helpers.asset_pack_path(server_bundle_js_file)
        File.join("public", public_subdir)
      else
        File.join(ReactOnRails.configuration.generated_assets_dir,
                  server_bundle_js_file)

      end
    end
```

---

Reviewed 9 of 17 files at r2, 9 of 15 files at r3, 4 of 8 files at r4, 6 of 7 files at r5, 3 of 4 files at r6, 3 of 4 files at r7, 1 of 1 files at r8, 13 of 13 files at r9.
Review status: all files reviewed at latest revision, 13 unresolved discussions, some commit checks broke.

---

*[lib/react_on_rails/server_rendering_pool/exec.rb, line 81 at r9](https://reviewable.io:443/reviews/shakacode/react_on_rails/844#-Kl0NQAcED7cReSIWBx2:-Kl0NQAdz0ePblpSIUz3:b-5osx1j) ([raw file](https://github.com/shakacode/react_on_rails/blob/47e7c86dc8e32e8c5cceaa360e6c36697e36ef43/lib/react_on_rails/server_rendering_pool/exec.rb#L81)):*
> ```Ruby
>         def create_js_context
>           manifest_key = ReactOnRails.configuration.server_bundle_js_file
>           server_js_file = "public/"+ActionController::Base.helpers.asset_pack_path(manifest_key)
> ```

rewriting this one, see main comment

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/844#-:-Kl0MvWG8NcBlkngcSmy:brokbwk)*
<!-- Sent from Reviewable.io -->

justin808 commented:
Reviewed 34 of 34 files at r10.
Review status: all files reviewed at latest revision, 13 unresolved discussions, some commit checks broke.

---

*[webpackConfigLoader.js, line 17 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/844#-KkLBqh38t7wEFetl3zl-r1-17:-KlDFCGqye4UJlCU6H7D:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/c49621a73705ce3dc79dfd5523b60f086247b4e2/webpackConfigLoader.js#L17)):*
<details><summary><i>Previously, ledestin (Dmitry Maksyoma) wrote…</i></summary><blockquote>

Consider extracting to `env.hotReloadingEnabled` to hide `TRUE` and `YES` details.
</blockquote></details>

Done.

---

*[lib/generators/react_on_rails/templates/base/base/client/webpack.config.js, line 29 at r5](https://reviewable.io:443/reviews/shakacode/react_on_rails/844#-KkLBqh38t7wEFetl3zg-r5-29:-KlDDedKrq3Ef1uyQsQY:bs7wupd) ([raw file](https://github.com/shakacode/react_on_rails/blob/e90bdb0cab87f9d902a5879d0ea3e89dbaa9c031/lib/generators/react_on_rails/templates/base/base/client/webpack.config.js#L29)):*
<details><summary><i>Previously, Conturbo (Jonathan Andrew Philippou) wrote…</i></summary><blockquote>

This may be a mistake, i'll check it again.
</blockquote></details>

Done.x

---

*[spec/dummy/client/webpack.client.rails.build.config.js, line 28 at r5](https://reviewable.io:443/reviews/shakacode/react_on_rails/844#-KkLBqh38t7wEFetl3zi-r5-28:-KlDFRXQDHKiFPElRUIf:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/e90bdb0cab87f9d902a5879d0ea3e89dbaa9c031/spec/dummy/client/webpack.client.rails.build.config.js#L28)):*
<details><summary><i>Previously, Conturbo (Jonathan Andrew Philippou) wrote…</i></summary><blockquote>

This was a change @robwise and I made bc not all leading slashes were added in all the right places. (Only in images).  Thus, we removed the leading slashes before images in webpack.common.config.js
</blockquote></details>

Done.

---

*[spec/dummy/client/webpack.common.config.js, line 11 at r5](https://reviewable.io:443/reviews/shakacode/react_on_rails/844#-Kkhh1U2FbuELuRF5Oa7-r5-11:-KlDF1yzAp7tUcJSBR97:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/e90bdb0cab87f9d902a5879d0ea3e89dbaa9c031/spec/dummy/client/webpack.common.config.js#L11)):*
<details><summary><i>Previously, Conturbo (Jonathan Andrew Philippou) wrote…</i></summary><blockquote>

We removed the leading slash because we already appended it as a trailing slash in webpack.client.rails.build.config.js as explained.
</blockquote></details>

Done.

---

*[spec/dummy/client/webpack.common.config.js, line 20 at r5](https://reviewable.io:443/reviews/shakacode/react_on_rails/844#-Kkhh1U2FbuELuRF5Oa7-r5-20:-KlDF0mclgxG1iTEqk5a:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/e90bdb0cab87f9d902a5879d0ea3e89dbaa9c031/spec/dummy/client/webpack.common.config.js#L20)):*
<details><summary><i>Previously, Conturbo (Jonathan Andrew Philippou) wrote…</i></summary><blockquote>

We removed the leading slash because we already appended it as a trailing slash in webpack.client.rails.build.config.js as explained.
</blockquote></details>

Done.

---

*[spec/dummy/client/webpack.server.rails.build.config.js, line 25 at r5](https://reviewable.io:443/reviews/shakacode/react_on_rails/844#-KkLBqh38t7wEFetl3zk-r5-25:-KlDFTCHidPWVwzRp5Wr:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/e90bdb0cab87f9d902a5879d0ea3e89dbaa9c031/spec/dummy/client/webpack.server.rails.build.config.js#L25)):*
<details><summary><i>Previously, Conturbo (Jonathan Andrew Philippou) wrote…</i></summary><blockquote>

This was a change @robwise and I made bc not all leading slashes were added in all the right places. (Only in images). Thus, we removed the leading slashes before images in webpack.common.config.js
</blockquote></details>

Done.

---

*[spec/dummy/spec/features/integration_spec.rb, line 161 at r5](https://reviewable.io:443/reviews/shakacode/react_on_rails/844#-KkmbsPRCypO2Z4oZ52g-r5-161:-KlDDp02g2uqjq7L5gA4:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/e90bdb0cab87f9d902a5879d0ea3e89dbaa9c031/spec/dummy/spec/features/integration_spec.rb#L161)):*
<details><summary><i>Previously, Conturbo (Jonathan Andrew Philippou) wrote…</i></summary><blockquote>

remove
</blockquote></details>

Done.

---

*[spec/dummy/spec/features/rails_context_spec.rb, line 33 at r8](https://reviewable.io:443/reviews/shakacode/react_on_rails/844#-KkvowjNjgOxVqcpUt8C-r8-33:-KlDF87DxalitHQSXGB5:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/553fe25d285b0102b1cf805618b75354bec4f245/spec/dummy/spec/features/rails_context_spec.rb#L33)):*
<details><summary><i>Previously, robwise (Rob Wise) wrote…</i></summary><blockquote>

commented-out code
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/844)*
<!-- Sent from Reviewable.io -->
